modules/assignment/src/main/java/com/intuit/wasabi/assignment/cache/impl/AssignmentsMetadataCacheRefreshTask.java

```diff
@@ -0,0 +1,85 @@
+/*******************************************************************************
+ * Copyright 2017 Intuit
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+
+package com.intuit.wasabi.assignment.cache.impl;
+
+import com.google.inject.Inject;
+import com.intuit.wasabi.assignment.cache.AssignmentMetadataCacheTimeService;
+import com.intuit.wasabi.assignment.cache.AssignmentsMetadataCache;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Date;
+
+/**
+ * This class is used to refresh assignment metadata cache.
+ *
+ */
+public class AssignmentsMetadataCacheRefreshTask implements Runnable {
+    private final Logger LOGGER = LoggerFactory.getLogger(AssignmentsMetadataCacheRefreshTask.class);
+
+    private AssignmentsMetadataCache metadataCache;
+    private AssignmentMetadataCacheTimeService timeService;
+
+    private Boolean refreshInProgress;
+    private Date lastRefreshTime;
+
+    @Inject
+    public AssignmentsMetadataCacheRefreshTask(AssignmentsMetadataCache metadataCache, AssignmentMetadataCacheTimeService timeService) {
```

Comment:
That strategy will incur performance degradation on the running assignment traffic so decided to do it in background...